src/modules/filters.js

```diff
@@ -134,26 +134,12 @@ import pako from "../libs/pako.js";
   */
 
   var FlateEncode = function(data) {
-    var arr = [];
+    var arr = new Uint8Array(data.length);
     var i = data.length;
-    var adler32;
-
     while (i--) {
       arr[i] = data.charCodeAt(i);
     }
-    adler32 = jsPDFAPI.adler32cs.from(data);
-    data = pako.deflate(arr);
-    arr = new Uint8Array(data.byteLength + 4);
-    arr.set(data, 0);
-    arr.set(
-      new Uint8Array([
-        adler32 & 0xff,
-        (adler32 >> 8) & 0xff,
-        (adler32 >> 16) & 0xff,
-        (adler32 >> 24) & 0xff
-      ]),
-      data.byteLength
-    );
+    arr = zlibSync(arr);
```

Comment:
Actually, this was a mistake by whoever did this last. The library was adding 4 extra bytes to the end and copying data for no reason. Funnily enough, the Adler32 checksum is actually invalid; it should be big endian, not little endian. The fact that most decompressors didn't care about the invalid extra bytes at the end is mostly luck.

Reply:
Also, I took a look at removing `adler32` and everything still seems to work, but I don't want to break anything so I haven't pushed it.

Reply:
>Actually, this was a mistake by whoever did this last. The library was adding 4 extra bytes to the end and copying data for no reason. The library used previously, pako, already added the checksum; this one does too. Funnily enough, the Adler32 checksum is actually invalid; it should be big endian, not little endian. The fact that most decompressors didn't care about the invalid extra bytes at the end is mostly luck.

Ah, thanks for explaining this to me :)

> Also, I took a look at removing adler32cs and everything still seems to work, but I don't want to break anything so I haven't pushed it.

I think you can safely push it ;) The only concern I have is that some users or plugins might have used the `adler32cs` function, because it was available on the jsPDF object. Since it never was intended to be public and is not documented as public, I would not really consider this a breaking change. So I think we can just remove it.